integrations/airflow/marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -70,33 +116,37 @@ def extract_on_complete(self, task_instance) -> StepMetadata:
 
         try:
             bigquery_job_id = self._get_xcom_bigquery_job_id(task_instance)
-            context['bigquery.job_id'] = bigquery_job_id
             if bigquery_job_id is None:
                 raise Exception("Xcom could not resolve BigQuery job id." +
                                 "Job may have failed.")
         except Exception as e:
             log.error(f"Cannot retrieve job details from BigQuery.Client. {e}",
                       exc_info=True)
-            context['bigquery.extractor.client_error'] = \
-                f"{e}: {traceback.format_exc()}"
             return StepMetadata(
                 name=get_job_name(task=self.operator),
-                context=context,
                 inputs=None,
-                outputs=None
+                outputs=None,
+                run_facets=[
+                    BigQueryErrorRunFacet(
+                        clientError=f"{e}: {traceback.format_exc()}",
+                        parserError=context.parser_error
+                    )
+                ]
             )
 
         inputs = None
         outputs = None
+        run_facets = []
         try:
             client = bigquery.Client()
             try:
                 job = client.get_job(job_id=bigquery_job_id)
-                job_properties_str = json.dumps(job._properties)
-                context['bigquery.job_properties'] = job_properties_str
+                props = job._properties
+
+                run_facets.append(self._get_output_statistics(props))
 
-                inputs = self._get_input_from_bq(job, context, source, client)
-                outputs = self._get_output_from_bq(job, source, client)
+                inputs = self._get_input_from_bq(props, source, client)
+                outputs = self._get_output_from_bq(props, source, client)
```

Comment:
I think the statistics facet makes sense attached to the run, given the billed bytes and cached properties. However, I do think it would make sense to attach the output rows to the output dataset- perhaps via a new facet? This is the way the spark integration posts stats and I think it would be good to have some consistency, if possible.

Reply:
Keeping current facet with billed bytes etc while moving output dataset statistics to output dataset facet makes sense :+1:

Reply:
Added this.